### PR TITLE
Kubelet migration to structured logs: cpumanager/{cpu_manager.go\fake_cpu_manager.go\policy_static.go)

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
pkg/kubelet/cm/cpumanager/cpu_manager.go NEEDS PR
pkg/kubelet/cm/cpumanager/fake_cpu_manager.go NEEDS PR
pkg/kubelet/cm/cpumanager/policy_static.go NEEDS PR

#### Which issue(s) this PR fixes:

xref #98976

#### Does this PR introduce a user-facing change?
```release-note
None
```
/cc ehashman

### CODE REVIEW

File: pkg/kubelet/cm/cpumanager/cpu_manager.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func NewManager(cpuPolicyName string, reconcilePeriod time.Duration, machineInfo
 		if err != nil {
 			return nil, err
 		}
-		klog.Infof("[cpumanager] detected CPU topology: %v", topo)
+		klog.InfoS("[cpumanager] detected CPU topology", "topology", topo)
```
Comment (ehashman) [inline pkg/kubelet/cm/cpumanager/cpu_manager.go:158]:
```suggestion
		klog.InfoS("detected CPU topology", "topology", topo)
```

Please remove all log line prefixes. (I won't comment on every line, only the ones that have other comments.)

File: pkg/kubelet/cm/cpumanager/cpu_manager.go [hunk 2]
```diff
@@ -419,15 +419,15 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
...
 			if cset.IsEmpty() {
 				// NOTE: This should not happen outside of tests.
-				klog.Infof("[cpumanager] reconcileState: skipping container; assigned cpuset is empty (pod: %s, container: %s)", pod.Name, container.Name)
+				klog.InfoS("[cpumanager] reconcileState: skipping container; assigned cpuset is empty", "pod", klog.KObj(pod), "containerName", container.Name)
 				failure = append(failure, reconciledContainer{pod.Name, container.Name, containerID})
 				continue
 			}
 
-			klog.V(4).Infof("[cpumanager] reconcileState: updating container (pod: %s, container: %s, container id: %s, cpuset: \"%v\")", pod.Name, container.Name, containerID, cset)
+			klog.V(4).InfoS("[cpumanager] reconcileState: updating container (pod: %s, container: %s, container id: %s, cpuset: \"%v\")", "pod", klog.KObj(pod), "containerName", container.Name, containerID, cset)
```
Comment (ehashman) [inline pkg/kubelet/cm/cpumanager/cpu_manager.go:427]:
```suggestion
			klog.V(4).InfoS("reconcileState: updating container", "pod", klog.KObj(pod), "containerName", container.Name, "containerID", containerID, "cpuSet", cset)
```

File: pkg/kubelet/cm/cpumanager/cpu_manager.go [hunk 3]
```diff
@@ -419,15 +419,15 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
...
 			err = m.updateContainerCPUSet(containerID, cset)
 			if err != nil {
-				klog.Errorf("[cpumanager] reconcileState: failed to update container (pod: %s, container: %s, container id: %s, cpuset: \"%v\", error: %v)", pod.Name, container.Name, containerID, cset, err)
+				klog.ErrorS(err, "[cpumanager] reconcileState: failed to update container", pod.Name, container.Name, "containerID", containerID, "cpuset", cset)
```
Comment (ehashman) [inline pkg/kubelet/cm/cpumanager/cpu_manager.go:430]:
```suggestion
				klog.ErrorS(err, "reconcileState: failed to update container", "pod", klog.KObj(pod), "containerName", container.Name, "containerID", containerID, "cpuSet", cset)
```

File: pkg/kubelet/cm/cpumanager/fake_cpu_manager.go [hunk 4]
```diff
@@ -70,12 +70,12 @@ func (m *fakeManager) State() state.Reader {
...
 
 func (m *fakeManager) GetCPUs(podUID, containerName string) cpuset.CPUSet {
-	klog.Infof("[fake cpumanager] GetCPUs(podUID: %s, containerName: %s)", podUID, containerName)
+	klog.InfoS("GetCPUs", "podUID", podUID, "containerName", containerName)
 	return cpuset.CPUSet{}
 }
 
 func (m *fakeManager) GetAllocatableCPUs() cpuset.CPUSet {
-	klog.Infof("[fake cpumanager] Get Allocatable Cpus")
+	klog.InfoS("Get Allocatable Cpus")
```
Comment (ehashman) [inline pkg/kubelet/cm/cpumanager/fake_cpu_manager.go:78]:
```suggestion
	klog.InfoS("Get Allocatable CPUs")
```

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 5]
```diff
@@ -218,23 +217,23 @@ func (p *staticPolicy) updateCPUsToReuse(pod *v1.Pod, container *v1.Container, c
...
 		// Call Topology Manager to get the aligned socket affinity across all hint providers.
 		hint := p.affinity.GetAffinity(string(pod.UID), container.Name)
-		klog.Infof("[cpumanager] Pod %v, Container %v Topology Affinity is: %v", format.Pod(pod), container.Name, hint)
+		klog.InfoS("[cpumanager] Topology Affinity", "pod", klog.KObj(pod), "containerName", container.Name, "affinity", hint)
 
 		// Allocate CPUs according to the NUMA affinity contained in the hint.
 		cpuset, err := p.allocateCPUs(s, numCPUs, hint.NUMANodeAffinity, p.cpusToReuse[string(pod.UID)])
 		if err != nil {
-			klog.Errorf("[cpumanager] unable to allocate %d CPUs (pod: %s, container: %s, error: %v)", numCPUs, format.Pod(pod), container.Name, err)
+			klog.ErrorS(err, "[cpumanager] unable to allocate CPUs", "numCPUs", numCPUs, "pod", klog.KObj(pod), "containerName", container.Name)
```
Comment (ehashman) [inline pkg/kubelet/cm/cpumanager/policy_static.go:236]:
```suggestion
			klog.ErrorS(err, "unable to allocate CPUs", "pod", klog.KObj(pod), "containerName", container.Name, "numCPUs", numCPUs)
```

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 6]
```diff
@@ -246,7 +245,7 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 }
 
 func (p *staticPolicy) RemoveContainer(s state.State, podUID string, containerName string) error {
-	klog.Infof("[cpumanager] static policy: RemoveContainer (pod: %s, container: %s)", podUID, containerName)
+	klog.InfoS("[cpumanager] static policy: RemoveContainer ", "podUID", podUID, "containerName", containerName)
```
Comment (ehashman) [inline pkg/kubelet/cm/cpumanager/policy_static.go:248]:
```suggestion
	klog.InfoS("static policy: RemoveContainer", "podUID", podUID, "containerName", containerName)
```

**Review by ehashman — COMMENTED** [review]
Just a few small comments, plus the log line prefix removals.

/milestone v1.21

**Comment by pacoxu** [comment]
Updated.

File: pkg/kubelet/cm/cpumanager/cpu_manager.go [hunk 7]
```diff
@@ -155,7 +155,7 @@ func NewManager(cpuPolicyName string, reconcilePeriod time.Duration, machineInfo
 		if err != nil {
 			return nil, err
 		}
-		klog.Infof("[cpumanager] detected CPU topology: %v", topo)
+		klog.InfoS("detected CPU topology", "topology", topo)
```
Comment (ehashman) [inline pkg/kubelet/cm/cpumanager/cpu_manager.go:158]: Ahhh I forgot to say this explicitly and I hate to block over this but we need to capitalize each log line. Like so:

```suggestion
		klog.InfoS("Detected CPU topology", "topology", topo)
```

Some of them already are, they all need to be consistently capitalized.

**Review by ehashman — COMMENTED** [review]
Very close to LGTM.

File: pkg/kubelet/cm/cpumanager/cpu_manager.go (hunk 7 above)
Comment (pacoxu) [reply pkg/kubelet/cm/cpumanager/cpu_manager.go]: Capitalized. ✅
😄

**Review by ehashman — COMMENTED** [review]
/lgtm

File: pkg/kubelet/cm/cpumanager/fake_cpu_manager.go [hunk 8]
```diff
@@ -32,36 +32,36 @@ type fakeManager struct {
...
 
 func (m *fakeManager) RemoveContainer(containerID string) error {
-	klog.Infof("[fake cpumanager] RemoveContainer (container id: %s)", containerID)
+	klog.InfoS("RemoveContainer", "containerID", containerID)
 	return nil
 }
 
 func (m *fakeManager) GetTopologyHints(pod *v1.Pod, container *v1.Container) map[string][]topologymanager.TopologyHint {
-	klog.Infof("[fake cpumanager] Get Container Topology Hints")
+	klog.InfoS("Get Container Topology Hints")
```
Comment (shubheksha) [inline pkg/kubelet/cm/cpumanager/fake_cpu_manager.go:59]: Nit
```suggestion
	klog.InfoS("Get container topology hints")
```

Not sure why this is capitalised

File: pkg/kubelet/cm/cpumanager/fake_cpu_manager.go [hunk 9]
```diff
@@ -32,36 +32,36 @@ type fakeManager struct {
...
 	return map[string][]topologymanager.TopologyHint{}
 }
 
 func (m *fakeManager) GetPodTopologyHints(pod *v1.Pod) map[string][]topologymanager.TopologyHint {
-	klog.Infof("[fake cpumanager] Get Pod Topology Hints")
+	klog.InfoS("Get Pod Topology Hints")
```
Comment (shubheksha) [inline pkg/kubelet/cm/cpumanager/fake_cpu_manager.go:64]:
```suggestion
	klog.InfoS("Get pod topology hints")
```

Same here

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 10]
```diff
@@ -256,7 +255,7 @@ func (p *staticPolicy) RemoveContainer(s state.State, podUID string, containerNa
 }
 
 func (p *staticPolicy) allocateCPUs(s state.State, numCPUs int, numaAffinity bitmask.BitMask, reusableCPUs cpuset.CPUSet) (cpuset.CPUSet, error) {
-	klog.Infof("[cpumanager] allocateCpus: (numCPUs: %d, socket: %v)", numCPUs, numaAffinity)
+	klog.InfoS("AllocateCpus", "numCPUs", numCPUs, "socket", numaAffinity)
```
Comment (shubheksha) [inline pkg/kubelet/cm/cpumanager/policy_static.go:258]:
```suggestion
	klog.InfoS("AllocateCPUs", "numCPUs", numCPUs, "socket", numaAffinity)
```

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 11]
```diff
@@ -353,15 +352,15 @@ func (p *staticPolicy) GetTopologyHints(s state.State, pod *v1.Pod, container *v
 	// kubelet restart, for example.
 	if allocated, exists := s.GetCPUSet(string(pod.UID), container.Name); exists {
 		if allocated.Size() != requested {
-			klog.Errorf("[cpumanager] CPUs already allocated to (pod %v, container %v) with different number than request: requested: %d, allocated: %d", format.Pod(pod), container.Name, requested, allocated.Size())
+			klog.ErrorS(nil, "CPUs already allocated to container with different number than request", "pod", klog.KObj(pod), "containerName", container.Name, "requested", requested, "allocated", allocated.Size())
```
Comment (shubheksha) [inline pkg/kubelet/cm/cpumanager/policy_static.go:355]:
```suggestion
			klog.ErrorS(nil, "CPUs already allocated to container with different number than request", "pod", klog.KObj(pod), "containerName", container.Name, "requestedSize", requested, "allocatedSize", allocated.Size())
```

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 12]
```diff
@@ -403,7 +402,7 @@ func (p *staticPolicy) GetPodTopologyHints(s state.State, pod *v1.Pod) map[strin
 		// kubelet restart, for example.
 		if allocated, exists := s.GetCPUSet(string(pod.UID), container.Name); exists {
 			if allocated.Size() != requestedByContainer {
-				klog.Errorf("[cpumanager] CPUs already allocated to (pod %v, container %v) with different number than request: requested: %d, allocated: %d", format.Pod(pod), container.Name, requestedByContainer, allocated.Size())
+				klog.ErrorS(nil, "CPUs already allocated to container with different number than request", "pod", klog.KObj(pod), "containerName", container.Name, "requested", requested, "requestedByContainer", requestedByContainer, "allocated", allocated.Size())
```
Comment (shubheksha) [inline pkg/kubelet/cm/cpumanager/policy_static.go:405]:
```suggestion
				klog.ErrorS(nil, "CPUs already allocated to container with different number than request", "pod", klog.KObj(pod), "containerName", container.Name, "requestedSize", requested, "requestedByContainer", requestedByContainer, "allocatedSize", allocated.Size())
```

**Review by shubheksha — CHANGES_REQUESTED** [review]
Very close :)

File: pkg/kubelet/cm/cpumanager/cpu_manager.go [hunk 13]
```diff
@@ -196,8 +196,8 @@ func NewManager(cpuPolicyName string, reconcilePeriod time.Duration, machineInfo
 }
 
 func (m *manager) Start(activePods ActivePodsFunc, sourcesReady config.SourcesReady, podStatusProvider status.PodStatusProvider, containerRuntime runtimeService, initialContainers containermap.ContainerMap) error {
-	klog.Infof("[cpumanager] starting with %s policy", m.policy.Name())
-	klog.Infof("[cpumanager] reconciling every %v", m.reconcilePeriod)
+	klog.InfoS("Starting", "policy", m.policy.Name())
```
Comment (mrunalp) [inline pkg/kubelet/cm/cpumanager/cpu_manager.go:199]: "Starting CPU manager"

File: pkg/kubelet/cm/cpumanager/fake_cpu_manager.go (hunk 8 above)
Comment (pacoxu) [reply pkg/kubelet/cm/cpumanager/fake_cpu_manager.go:59]: @ehashman

**Comment by pacoxu** [comment]
@shubheksha @mrunalp 
Updated per your comments. Would you take a look again?

**Review by ehashman — COMMENTED** [review]
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**